client/jetpack-connect/authorize.js

```diff
@@ -888,7 +889,14 @@ export class JetpackAuthorize extends Component {
 		const productSlug = searchParams.get( 'productSlug' );
 		const siteSlug = searchParams.get( 'fromSiteSlug' );
 		const product = getProductFromSlug( productSlug );
-		const productName = getJetpackProductDisplayName( product );
+		let productName = '';
+
+		if ( product?.product_name ) {
+			productName = getJetpackProductDisplayName( product );
+		} else {
+			productName = getPlan( productSlug )?.getTitle();
```

Comment:
Could this be turned into a utility type function like `getJetpackProductDisplayName`? Might be useful elsewhere

Reply:
Nice idea - I'll look to add it in the same place as `getJetpackProductDisplayName `. Maybe we call it `getJetpackProductOrPlanDisplayName `?